### PR TITLE
fix(item preview): adding item preview component

### PR DESCRIPTION
**Pull request by anthony-liddle**
## Goal

Adding functionality to display an item preview

## Todos:
- [x] Add item deriver
- [x] Display item preview

## Implementation Decisions

Screen Shot 2021-09-28 at 1 10 39 PM

### CODE REVIEW

File: src/actions.js [hunk 1]
```diff
@@ -14,9 +14,11 @@ export const SAVE_TO_POCKET_FAILURE = 'SAVE_TO_POCKET_FAILURE'
...
 export const SAVE_TO_POCKET_COMPLETE = 'SAVE_TO_POCKET_COMPLETE'
 
-export const ARCHIVE_ITEM_REQUEST = 'ARCHIVE_ITEM_REQUEST'
-export const ARCHIVE_ITEM_SUCCESS = 'ARCHIVE_ITEM_SUCCESS'
-export const ARCHIVE_ITEM_FAILURE = 'ARCHIVE_ITEM_FAILURE'
+export const UPDATE_ITEM_PREVIEW = 'UPDATE_ITEM_PREVIEW'
+
+// export const ARCHIVE_ITEM_REQUEST = 'ARCHIVE_ITEM_REQUEST'
+// export const ARCHIVE_ITEM_SUCCESS = 'ARCHIVE_ITEM_SUCCESS'
+// export const ARCHIVE_ITEM_FAILURE = 'ARCHIVE_ITEM_FAILURE'
```
Comment (anthony-liddle) [inline src/actions.js:21]: Commenting out for now. Will be stripped later

File: src/common/helpers.js [hunk 2]
```diff
@@ -34,10 +34,10 @@ export function getAPIUrl() {
...
 }
 
-export function getSlug(url) {
-  const match = url.match(/^(?:https:\/\/getpocket.com\/explore\/)(.+[^/])/i)
-  return match[1]
-}
+// export function getSlug(url) {
+//   const match = url.match(/^(?:https:\/\/getpocket.com\/explore\/)(.+[^/])/i)
+//   return match[1]
+// }
```
Comment (anthony-liddle) [inline src/common/helpers.js:40]: I don't think we're using this. To be stripped later.

File: src/connectors/doorhanger/doorhanger.js [hunk 3]
```diff
@@ -24,10 +26,10 @@ const doorhangerStyle = css`
...
     right: 10px;
     transform: translateY(-150%);
-    display: flex;
-    justify-content: flex-start;
-    align-items: center;
-    align-content: center;
+    /* display: flex; */
+    /* justify-content: flex-start; */
+    /* align-items: center; */
+    /* align-content: center; */
```
Comment (anthony-liddle) [inline src/connectors/doorhanger/doorhanger.js:32]: This may end up uncommented, but for now it was causing all sorts of display issues

Comment (anthony-liddle) [reply src/connectors/doorhanger/doorhanger.js:32]: And these were the culprit styles preventing it from displaying on other pages.

Comment (wtfluckey) [reply src/connectors/doorhanger/doorhanger.js:32]: 🙌 Thank you for digging into this one! It was driving me up the wall!

File: src/pages/background/postSave.js [hunk 4]
```diff
@@ -15,13 +17,27 @@ export async function saveSuccess(tabId, payload) {
...
+–––––––––––––––––––––––––––––––––––––––––––––––––– */
+async function getItemPreview(tabId, payload) {
+  const item = await deriveItemData(payload)
+
+  chrome.tabs.sendMessage(tabId, {
+    action: UPDATE_ITEM_PREVIEW,
+    payload: { item },
+  })
+}
+
```
Comment (wtfluckey) [inline src/pages/background/postSave.js:40]: Nice!

**Review by wtfluckey — APPROVED** [review]

**Pull request merged**